sklearn/mixture/tests/test_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,264 @@
+
+    # Asser that we get perfect clustering
+    ari = adjusted_rand_score(y, gmIC.fit_predict(X))
+    assert_allclose(ari, 1)
+
+    # test AIC
+    gmIC = GaussianMixtureIC(max_components=5, criterion="aic", **kws)
+    gmIC.fit(X, y)
+    n_components = gmIC.n_components_
+
+    # AIC gets the number of components wrong
+    assert_equal(n_components >= 1, True)
+    assert_equal(n_components <= 5, True)
+
+
+def test_two_class_parallel():
+    _test_two_class(n_jobs=2)
+
+
+def test_five_class():
+    """
+    Easily separable five gaussian problem.
+    """
+    np.random.seed(1)
+
+    n = 100
+    mus = [[i * 5, 0] for i in range(5)]
+    cov = np.eye(2)  # balls
+
+    X = np.vstack([np.random.multivariate_normal(mu, cov, n) for mu in mus])
+
+    # test BIC
+    gmIC = GaussianMixtureIC(
+        min_components=3,
+        max_components=10,
+        covariance_type="all",
+        criterion="bic",
+    )
+    gmIC.fit(X)
+    assert_equal(gmIC.n_components_, 5)
+
+    # test AIC
+    gmIC = GaussianMixtureIC(
+        min_components=3,
+        max_components=10,
+        covariance_type="all",
+        criterion="aic",
+    )
+    gmIC.fit(X)
+    # AIC fails often so there is no assertion here
+    assert_equal(gmIC.n_components_ >= 3, True)
+    assert_equal(gmIC.n_components_ <= 10, True)
+
+
+def test_covariances():
+    """
+    Easily separable two gaussian problem.
+    """
+    np.random.seed(1)
+
+    n = 100
+    mu1 = [-10, 0]
+    mu2 = [10, 0]
+
+    # Spherical
+    cov1 = 2 * np.eye(2)
+    cov2 = 2 * np.eye(2)
+
+    X1 = np.random.multivariate_normal(mu1, cov1, n)
+    X2 = np.random.multivariate_normal(mu2, cov2, n)
+
+    X = np.concatenate((X1, X2))
+
+    gmIC = GaussianMixtureIC(min_components=2, covariance_type="all")
+    gmIC.fit(X)
+    assert_equal(gmIC.covariance_type_, "spherical")
+
+    # Diagonal
+    np.random.seed(10)
+    cov1 = np.diag([1, 1])
+    cov2 = np.diag([2, 1])
+
+    X1 = np.random.multivariate_normal(mu1, cov1, n)
+    X2 = np.random.multivariate_normal(mu2, cov2, n)
+
+    X = np.concatenate((X1, X2))
+
+    gmIC = GaussianMixtureIC(max_components=2, covariance_type="all")
+    gmIC.fit(X)
+    assert_equal(gmIC.covariance_type_, "diag")
+
+    # Tied
+    cov1 = np.array([[2, 1], [1, 2]])
+    cov2 = np.array([[2, 1], [1, 2]])
+
+    X1 = np.random.multivariate_normal(mu1, cov1, n)
+    X2 = np.random.multivariate_normal(mu2, cov2, n)
+
+    X = np.concatenate((X1, X2))
+
+    gmIC = GaussianMixtureIC(max_components=2, covariance_type="all")
+    gmIC.fit(X)
+    assert_equal(gmIC.covariance_type_, "tied")
+
+    # Full
+    cov1 = np.array([[2, -1], [-1, 2]])
+    cov2 = np.array([[2, 1], [1, 2]])
+
+    X1 = np.random.multivariate_normal(mu1, cov1, n)
+    X2 = np.random.multivariate_normal(mu2, cov2, n)
+
+    X = np.concatenate((X1, X2))
+
+    gmIC = GaussianMixtureIC(max_components=2, covariance_type="all")
+    gmIC.fit(X)
+    assert_equal(gmIC.covariance_type_, "full")
```

Comment:
You should be able to factor some this code out by [parametrising the test](https://docs.pytest.org/en/stable/example/parametrize.html) on `cov1`, `cov2` and the assumed values of `gmIC.covariance_type_`.